pkg/machine/wsl/machine.go

```diff
@@ -136,9 +136,56 @@ Wants=network-online.target podman.socket
 ExecStart=/usr/bin/sleep infinity
 `
 
-const lingerSetup = `mkdir -p /home/[USER]/.config/systemd/[USER]/default.target.wants
-ln -fs /home/[USER]/.config/systemd/[USER]/linger-example.service \
-       /home/[USER]/.config/systemd/[USER]/default.target.wants/linger-example.service
+const lingerSetup = `mkdir -p /home/[USER]/.config/systemd/user/default.target.wants
+ln -fs /home/[USER]/.config/systemd/user/linger-example.service \
+       /home/[USER]/.config/systemd/user/default.target.wants/linger-example.service
+`
+
+const bindMountSystemService = `
+[Unit]
+Description=Bind mount for system podman sockets
+After=podman.socket
+
+[Service]
+RemainAfterExit=true
+Type=simple
+# Ensure user services can register sockets as well
+ExecStartPre=mkdir -p -m 777 /mnt/wsl/podman-sockets/%[1]s
+ExecStartPre=touch /mnt/wsl/podman-sockets/%[1]s/podman-root.sock
+ExecStart=mount --bind %%t/podman/podman.sock /mnt/wsl/podman-sockets/%[1]s/podman-root.sock
+ExecStop=umount /mnt/wsl/podman-sockets/%[1]s/podman-root.sock
+`
+
+const bindMountUserService = `
+[Unit]
+Description=Bind mount for podman sockets
+After=podman.socket
+
+[Service]
+RemainAfterExit=true
+Type=simple
+# Consistency with system service (supports racing)
+ExecStartPre=mkdir -p -m 777 /mnt/wsl/podman-sockets/%[1]s
+ExecStartPre=touch /mnt/wsl/podman-sockets/%[1]s/podman-user.sock
+# Relies on /etc/fstab entry for user mounting
+ExecStart=mount /mnt/wsl/podman-sockets/%[1]s/podman-user.sock
+ExecStop=umount /mnt/wsl/podman-sockets/%[1]s/podman-user.sock
+`
+
+const bindMountFsTab = `/run/user/1000/podman/podman.sock /mnt/wsl/podman-sockets/%s/podman-user.sock none noauto,user,bind,defaults 0 0
```

Comment:
why does this use fstab at all? By the time fstab tries to mount it the files will not be there anyway so it will fail and only be useful for the mount command in the unit.
But then you might as well just type the full mount command into the unit and skip the fstab part.


Also you can create .mount units with systemd then you could just fire up with the depedncy management of systemd at the right time.

Reply:
@Luap99 it needs fstab because one of the services has a dependency on a `systemd --user` process, so has to run as non-root, and root is normally required to issue mounts unless there is a pre-declared fstab entry (user option). It's also marked as noauto to keep it from being picked up by anything other than this mount.

The only reason this wasn't done as mount unites was just to keep everything as just one unit file [per root/user] (in addition to the mount it also needs to precreate the tmpfs directory structure).  Unfortunately this has to be duplicated in both units since systemd does not support dependencies that cross user/root.

Reply:
Ah right, that makes sense then.